lib/prometheus/que_stats.rb

```diff
@@ -54,23 +83,17 @@ def execute
 Yabeda.configure do
   group :que do
     scheduled_jobs = gauge :jobs_scheduled_total, comment: 'Que Jobs to be executed'
-    workers = gauge :workers_total,  comment: 'Que Workers running'
-
-    collect do
-      Prometheus::QueStats.job_stats.each do |stats|
-        scheduled_jobs.set({ job: stats.fetch('job_class') }, stats.fetch('count'))
-      end
+    workers = gauge :workers_total, comment: 'Que Workers running'
 
-      Prometheus::QueStats.job_stats(%q[(args->0->>'retries')::integer > 0]).each do |stats|
-        scheduled_jobs.set({ job: stats.fetch('job_class'), type: 'retry' }, stats.fetch('count'))
-      end
+    set_stats = ->(type, stats) { scheduled_jobs.set({ job: stats.fetch('job_class') }.merge(type), stats.fetch('count')) }
 
-      Prometheus::QueStats.job_stats('run_at < now()').each do |stats|
-        scheduled_jobs.set({ job: stats.fetch('job_class'), type: 'scheduled' }, stats.fetch('count'))
-      end
+    collect do
+      set_stats_all = set_stats.curry.call({})
```

Comment:
Yes `type` is definitely a `Hash` ... this is unexpected. If the reader does not know how `curry` works, one can be confused

Reply:
I am that confused reader unfortunately 🙈 Sorry 🙏 

And what is `curry` ? 😱

Reply:
@Martouta It is a feature of functional programming. The best ever post I read about it was on SO https://stackoverflow.com/questions/36314/what-is-currying

Reply:
Oh, I see. Thanks! ❤️ I got to know about this functional programming feature thanks to Didier when he taught us a bit about Elm, but I didn't know that this feature had the name _curry_ 😄

Reply:
Well, people not knowing how a core Ruby method works should not be an excuse for not using it, no? However, I did think about doing this completely different BTW, for stylistic reasons. We could have a proper class that fetches the count and sets the stats, with proper methods instead using procs and currying.

Reply:
@guicassolato The request is not to stop using it, the request is changing the name `type` to make more clear what it is 😄 

https://github.com/3scale/zync/pull/334/files#r429113867 & https://github.com/3scale/zync/pull/334/files#r429113547 are very related